client/state/selectors/can-any-site-have-plugins.ts

```diff
@@ -0,0 +1,25 @@
+import { createSelector } from '@automattic/state-utils';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import getSites from 'calypso/state/selectors/get-sites';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import type { AppState } from 'calypso/types';
+
+import 'calypso/state/ui/init';
+
+/**
+ * Return true if is user is able to use plugins on any of their sites.
+ *
+ * @param state Global state tree
+ */
+
+export const canAnySiteHavePlugins = createSelector(
+	( state: AppState ): boolean =>
+		getSites( state ).some(
+			( site ) =>
+				site &&
+				isJetpackSite( state, site.ID ) &&
+				canCurrentUser( state, site.ID, 'manage_options' ) &&
+				site.visible
+		),
+	( state ) => [ state.sites.items, state.currentUser.capabilities ]
+);
```

Comment:
Nice, I hadn't seen that one.

The problem is that `getSelectedOrAllSitesJetpackCanManage()` doesn't always check all sites, sometimes it checks just the selected site. That doesn't make a difference right now—the new selector's value is currently only used when there is no selected site. But that might not always be the case.
I'd rather keep this copy-paste so that the two selectors, which have different semantics, can diverge over time.